test/loading.jl

```diff
@@ -4,10 +4,8 @@ using Base.Test
 
 @test @__LINE__ == 5
 
-@test_throws ArgumentError Core.include("test_sourcepath.jl\0")
-
 include("test_sourcepath.jl")
-thefname = "the fname!//\\&\0\1*"
+thefname = "the fname!//\\&\1*"
```

Comment:
We never actually support filenames (either real or fake ones) that are not NUL-terminated strings. E.g., for real filenames, the `jl_stats` call assumes NUL termination, for both real and fake ones, every functions that uses `jl_filename` assumes it is NUL terminated. It might be possible to go through everything and fix them but it would be much more work and I highly doubt supporting embeded NUL or non-NUL terminating strings as filename is useful.